### DISCUSSION TITLE
https://plugin-components.vuejs.press/zh/guide/media/p-d-f.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#692] [Announcements]
# https://plugin-components.vuejs.press/zh/guide/media/p-d-f.html

PDF 预览组件。 你可以使用此组件在 Markdown 中嵌入 PDF 阅读器。 示例 默认 PDF 阅读器 禁用工具栏且初始页面为第二页的阅读器 PDFJS 查看器 并非所有浏览器都支持嵌入 PDF 查看器 (如: 现在没有移动浏览器支持此功能)，因此我们添加了对 PDFJS 查看器的支持。 由于 PDFJS 查看器 很大，你需要从 Github ...

https://plugin-components.vuejs.press/zh/guide/media/p-d-f.html

**Comment by 006hjy** [comment]
有没有办法让pdf默认适应宽度

**Reply by Mister-Hope** [reply]
没有，这是浏览器的原生功能。

**Comment by myorange2009** [comment]
为什么我嵌入案例代码正常，用自己的链接却是打开页面直接下载
哪里没有设置对？

**Reply by myorange2009** [reply]
解决了，是我COS存储的问题

**Comment by dream-oyh** [comment]
为什么我的pdf组件没有全屏按钮，我的配置如下：
theme.ts里：
import { hopeTheme } from "vuepress-theme-hope";
export default hopeTheme({
plugins:{
   components: {
			components: ["SiteInfo", "PDF"],
			// componentOptions: {
			// 	pdf: {
			// 		pdfjs: "/pdfjs-4.0.189-dist",
			// 	},
			// },
		},
}
})
在markdown插入pdf文件时：
<PDF url="/pdf/example.pdf" />
其中url是我在public文件夹下的相对路径，pdf是可以正常读取并打开的，但是右下角没有全屏按钮，我在markdown中插入pdf文件时也并没有打开noFullscreen

目前已经尝试：
我怀疑是缓存的问题，因为我之前有设置过自定义版本的pdfjs，但是删除.temp文件夹并重启博客后，依然无效

**Reply by dream-oyh** [reply]
漏打上来了，markdown插入pdf文件的语句是这样的
<PDF url="/pdf/delete_and_accept.pdf" />

**Reply by Mister-Hope** [reply]
你用的是最新版么？

**Comment by tsing-dong** [comment]
<PDF url="//theme-hope-assets.vuejs.press/files/20240705.pdf" /> markdown 中这样引入 文件放在 public files 文文件夹下 展示不成功 这种引入方式有问题吗

**Comment by yutian81** [comment]
## 移动端 PDF 无法预览

我使用的主题版本：`"vuepress-theme-hope": "2.0.0-rc.99"`

已按教程在 [PDFjs 官方仓库](https://github.com/mozilla/pdf.js)下载了最新的 release 包，版本：`v5.4.449`

并将该包解压存放到 `src/.vuepress/public/pdfjs/` 下，其中包含 `build` `web` 两个文件夹

主题配置 `theme.js` 组件配置启用了 PDF：

```js
  hostname: "https://wiki.notett.com",
  author: {
    name: "Daoyi Real Estate",
    url: "https://wiki.notett.com",
  },

// 其他配置

    components: {
      components: ["SiteInfo", "VidStack", "ArtPlayer", "BiliBili", "Share", "PDF", "Badge", "VPCard", "VPBanner",],
      componentOptions: {
        pdf: {pdfjs: "/pdfjs",},
      }
    },
```

访问以下地址可以打开示例PDF，说明 PDFjs 组件的路径没错：

```
https://wiki.notett.com/pdfjs/web/viewer
```

MD 中语法如下：

```
<PDF viewer="true" url="https://r2qq.24811213.xyz/dichan/00精品-商业定位-2023运河邻里中心项目定位及品牌规划方案.pdf" />
```

`r2qq.24811213.xyz` 是我的 CF R2存储桶自定义域名，存储桶设置中，已经允许跨域 `https://*.notett.com`

**现在的问题是：所有组件都正常，唯独 PDF 移动端无法预览**

作者的示例页面，手机预览PDF是没问题的：[点击查看](https://theme-hope.vuejs.press/zh/guide/component/built-in.html#pdf)

但我的就是不行：[点击查看](https://wiki.notett.com/res/%E5%8F%82%E8%80%83%E6%8A%A5%E5%91%8A/%E5%89%8D%E7%AD%96%E5%AE%9A%E4%BD%8D/2023-%E8%BF%90%E6%B2%B3%E9%82%BB%E9%87%8C%E4%B8%AD%E5%BF%83%E9%A1%B9%E7%9B%AE%E5%AE%9A%E4%BD%8D%E5%8F%8A%E5%93%81%E7%89%8C%E8%A7%84%E5%88%92%E6%96%B9%E6%A1%88.html)

**我实在不知道哪里错了，特来求助！**

**Comment by Mister-Hope** [comment]
也许你可能需要把路径encode一下。

**Reply by yutian81** [reply]
这样吗？我试试看再来回复

**Reply by yutian81** [reply]
> 也许你可能需要把路径encode一下。

```
<PDF
  url="https://r2qq.24811213.xyz/dichan/00%E7%B2%BE%E5%93%81-%E5%95%86%E4%B8%9A%E5%AE%9A%E4%BD%8D-2023%E8%BF%90%E6%B2%B3%E9%82%BB%E9%87%8C%E4%B8%AD%E5%BF%83%E9%A1%B9%E7%9B%AE%E5%AE%9A%E4%BD%8D%E5%8F%8A%E5%93%81%E7%89%8C%E8%A7%84%E5%88%92%E6%96%B9%E6%A1%88.pdf"
  viewer="true"
/>
```

还是不行

[页面地址](https://wiki.notett.com/res/%E5%8F%82%E8%80%83%E6%8A%A5%E5%91%8A/%E5%89%8D%E7%AD%96%E5%AE%9A%E4%BD%8D/2023-%E8%BF%90%E6%B2%B3%E9%82%BB%E9%87%8C%E4%B8%AD%E5%BF%83%E9%A1%B9%E7%9B%AE%E5%AE%9A%E4%BD%8D%E5%8F%8A%E5%93%81%E7%89%8C%E8%A7%84%E5%88%92%E6%96%B9%E6%A1%88.html)
